src/openai/_models.py

```diff
@@ -363,7 +363,15 @@ class GenericModel(BaseGenericModel, BaseModel):
 
 
 if PYDANTIC_V2:
-    from pydantic import TypeAdapter
+    from functools import lru_cache
+
+    from pydantic import TypeAdapter as PyTypeAdapter
+
+    def TypeAdapter(type_: type[_T]) -> PyTypeAdapter[_T]:
+        return PyTypeAdapter(type_)
+    
+    if not TYPE_CHECKING:
+        TypeAdapter = lru_cache(TypeAdapter)
```

Comment:
nit: I think this conveys the intent of this code more clearly (and you can add the `lru_cache` import at the top of the file), thoughts?

```suggestion
    if TYPE_CHECKING:
      from pydantic import TypeAdapter
    else:
      from pydantic import TypeAdapter as _TypeAdapter
      
      TypeAdapter = lru_cache(TypeAdapter)   
```

Reply:
I agree, It has been modified.